### PR TITLE
travis publish.sh allow releases from branches

### PR DESCRIPTION
**Pull request by pavolloffay**
This should allow releasing from any `digit.digit.digit` branch. This uses `0.0.0-RC0` pattern for releases. 

To release switch to the `0.31.0` branch tag `release-0.31.0-RC1` and push to upstream.

cc @tedsuo

### CODE REVIEW

**Comment by tedsuo** [comment]
LGTM

File: travis/publish.sh [hunk 1]
```diff
@@ -35,12 +35,12 @@ is_pull_request() {
...
-is_travis_branch_master() {
-  if [ "${TRAVIS_BRANCH}" = master ]; then
-    echo "[Publishing] Travis branch is master"
+is_travis_branch_master_or_release() {
+  if [[ "${TRAVIS_BRANCH}" == "master" || "${TRAVIS_BRANCH}" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+    echo "[Publishing] Travis branch is ${TRAVIS_BRANCH}"
     return 0
   else
-    echo "[Not Publishing] Travis branch is not master"
+    echo "[Not Publishing] Travis branch is not master or 0.0.0"
```
Comment (sjoerdtalsma) [inline travis/publish.sh]: nit: change 0.0.0 to _a release branch_  ?

Comment (pavolloffay) [reply travis/publish.sh]: I want to make it explicit how a release branch should look like.

**Comment by pavolloffay** [comment]
`safe_checkout_master()` has to be modified too. I will open follow-up PR.

File: travis/publish.sh (hunk 1 above)
Comment (sjoerdtalsma) [reply travis/publish.sh]: So a release branch should be 0.0.0 ? 😉

Comment (yurishkuro) [reply travis/publish.sh]: `s/or 0.0.0/a release branch #.#.#/`

**Pull request merged**